src/App/App.xaml.cs

```diff
@@ -221,7 +221,7 @@ private async Task<bool> CheckShouldSwitchActiveUserAsync(PasswordlessRequestNot
             }
 
             var notificationUserEmail = await _stateService.GetEmailAsync(notification.UserId);
-            await Device.InvokeOnMainThreadAsync(async () =>
+            Device.BeginInvokeOnMainThread(async () =>
```

Comment:
You need to add a try...catch inside otherwise if an exception occurs inside, it'll crash the app.
OTOH, given that you have removed the `await` now the lock will last less cause the call to the main thread won't wait until the inside finishes (unless I'm seeing something wrong). So you could potentially have the main thread block running and another `CheckShouldSwitchActiveUserAsync(...)` called (you can test this by having a delay of 10 seconds inside the main thread block and calling again the check)